README.md

```diff
@@ -442,6 +442,40 @@ Notice a few things about our single-file component:
 
 Try running `npm run build` and open up `index.html` to see the result!
 
+# Use Decorators to define a component
+
+Components can also be defined using [Decorators](https://www.typescriptlang.org/docs/handbook/decorators.html). With the help of two additional packages ([vue-class-component](https://github.com/vuejs/vue-class-component) and [vue-property-decorator](https://github.com/kaorun343/vue-property-decorator)), the definition of our component can be simplified as following:
+
+```ts
+@Component
+export default class HelloDecorator extends Vue {
+    @Prop() public name: string;
+    @Prop() public initialEnthusiasm: number;
+    enthusiasm = this.initialEnthusiasm;
+    increment() {
+        this.enthusiasm++;
+    }
+    decrement() {
+        if (this.enthusiasm > 1) {
+            this.enthusiasm--;
+        }
+    }
+    get exclamationMarks(): string {
+        return Array(this.enthusiasm + 1).join('!');
+    }
+}
+```
+
+Instead of using `Vue.extend` to define our component, we create a TypeScript class extending `Vue` and prefix it with the `@Component` decorator from the `vue-class-component` package.

```

Comment:
If you make your imports explicit, you don't have to mention where the decorators originate from.